pkg/api/v1/routes/handlers_test.go

```diff
@@ -306,6 +314,168 @@ func TestAddServer(t *testing.T) {
 	}
 }
 
+// nolint:gocyclo // cyclomatic tests are cyclomatic
+func TestAddServerRollback(t *testing.T) {
```

Comment:
It's definitely out-of-scope for this PR, but reading these tests convinces me that this is not the right layer for these tests, because we're testing handlers behavior by testing HTTP requests and responses instead of directly invoking the `Handlers` object.

Reply:
Can you please help provide more explanation?

Do you suggestion we mock [server service client](https://github.com/metal-toolbox/hollow-serverservice/blob/9089164ea19a188d3e5eb79d56892e23d9b2f121/pkg/api/v1/server_service.go#L28-L64) to get more test coverage?